schemas/derivatives/SCHEMA.md

```diff
@@ -0,0 +1,69 @@
+# Derivatives (Perps / Options) Schema
+
+This is the OpenBlock Labs standard schema for derivatives protocols (options and perps).
+
+## Schema
+
+### Pools
+
+**Description:** Pools in the protocol (one entry for each token).
+
+| Property            | Description                                        | Type   |
+|---------------------|----------------------------------------------------|--------|
+| chain_id            | Standard chain id.                                 | number |
+| creation_timestamp  | The timestamp of the block the pool was created on.| number |
+| creation_block_number | The block number this pool was created on.       | number |
+| pool_address        | The contract address of the pool.                  | string |
+| token_index         | The index of the token in the smart contract (one row for each token in a pool). | number |
+| token_address       | The contract address of the token.                 | string |
+| token_symbol        | The symbol of the token.                           | string |
+
+### LP Snapshot
+
+**Description:** Liquidity providers snapshot (one entry for each token in a pool).
+
+| Property          | Description                                           | Type   |
+|-------------------|-------------------------------------------------------|--------|
+| timestamp         | The timestamp of the snapshot.                        | number |
+| chain_id          | Standard chain id.                                    | number |
+| pool_address      | The address of the pool.                              | string |
+| lp_address        | The address of the liquidity provider.                | string |
+| token_index       | The index of the token in the smart contract.         | number |
+| token_address     | The address of the token provided as liquidity.       | string |
+| token_symbol      | The symbol of the token.                              | string |
+| token_amount      | The amount of tokens, decimal normalized.             | number |
+| token_amount_usd  | (Optional) The amount of the provided token in USD.   | number |
+| liquid_token_amount | The amount of tokens supplied that are liquid and usable by users of the protocol, decimal normalized. | number |
```

Comment:
liquid_token_amount - is this something like deposited collateral - collateral that are used in active positions

Reply:
This was the name I came up with for what the modelers called `amount_useful`. Open to suggestions on this one, I am not as keyed in on this as you

Reply:
if it is something like

`deposited collateral - collateral that are used in active positions`

its not that easy to calculate via events IMO because it fluctuates over time

Reply:
that's the thing - most Perps are orderbook, so you'd need to distinguish makers from takers there. And maker_useful_amount is what is within depth. We are working on a MM methodology with it's own schema. Same applies for LPs in AMMs, but rather rare in Perps. The only one I know is RabbitX on Starknet and it's not ingested yet on our side, because they haven't deployed the contracts there. We have proxied in the past for takers is deposit_amount - OI_value.

Reply:
I agree with @0xroll here. But @lylprg I am also good to leave it in here for now until we get feedback if you think it is a field that is needed (`liquid_token_amount` in the LP Snapshot)